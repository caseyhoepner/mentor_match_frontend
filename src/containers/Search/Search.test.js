import React from 'react';
import { shallow } from 'enzyme';
import { Search, mapStateToProps, mapDispatchToProps } from './';

describe('Search', () => {
  let wrapper;
  let mockState = { 
    localeSelected: '',
    search: '',
    favClicked: false,
    allMentorsClicked: false,
    preferencesClicked: false
  }
  let mockSetLocale = jest.fn()
  let mockSetSearch = jest.fn()
  
  beforeEach(() => {
    wrapper = shallow(<Search />)
  });
  
  describe('Search container', () => {
    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should update redux state with locale', () => {
      wrapper = shallow(<Search setLocale={mockSetLocale} />)
      let mockLocaleEvent = { target: { name: 'localeSelected', value: 'Denver' } }
      
      wrapper.instance().handleChange(mockLocaleEvent)
      expect(mockSetLocale).toHaveBeenCalled()
    });

    it('should update redux state with search terms', () => {
      let mockSearchEvent = { target: { name: 'search', value: 'skwijb' } }
      wrapper = shallow(<Search setSearch={mockSetSearch} />)

      wrapper.instance().handleChange(mockSearchEvent)
      expect(mockSetSearch).toHaveBeenCalled()
    });

    it('should toggle state if the Favorite button is clicked', () => {
      let mockToggleEvent = { target: { name: 'favClicked' } }

      wrapper.instance().toggleClicked(mockToggleEvent)
      expect(wrapper.state().favClicked).toEqual(true)
    });

    it('should toggle state if the All Mentors button is clicked', () => {
      let mockToggleEvent = { target: { name: 'allMentorsClicked' } }

      wrapper.instance().toggleClicked(mockToggleEvent)
      expect(wrapper.state().allMentorsClicked).toEqual(true)
    });

    it('should toggle state if the Preferences button is clicked', () => {
      let mockToggleEvent = { target: { name: 'preferencesClicked' } }

      wrapper.instance().toggleClicked(mockToggleEvent)
      expect(wrapper.state().preferencesClicked).toEqual(true)
    });
  });

  describe('mapStateToProps function', () => {
    it('should return an object with the seach params currently assigned', () => {
      const mockState = { 
        locale: 'Denver',
        searchTerm: 'booyah',
        showingAllMentors: true, 
        somethingElse: 'wooooooo' 
      }
      const expected = { 
        locale: 'Denver',
        searchTerm: 'booyah',
        showingAllMentors: true 
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  });

  describe('mapDispatchToProps function', () => {
    const mockDispatch = jest.fn()
    let mappedProps;

    beforeEach(() => {
      mappedProps = mapDispatchToProps(mockDispatch)
    });

    it('should call dispatch when setLocale is called', () => {
      mappedProps.setLocale()
      expect(mockDispatch).toHaveBeenCalled()
    });

    it('should call dispatch when setSearch is called', () => {
      mappedProps.setSearch()
      expect(mockDispatch).toHaveBeenCalled()
    });

    it('should call dispatch when toggleShowingMentors is called', () => {
      mappedProps.toggleShowingMentors(true)
      expect(mockDispatch).toHaveBeenCalled()
    });
  });
});