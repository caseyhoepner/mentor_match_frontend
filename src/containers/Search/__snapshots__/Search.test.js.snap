// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search container matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="s-search-and-pref-container"
      >
        <div
          className="s-search-and-buttons"
        >
          <div
            className="s-search-inputs-container"
          >
            <div
              className="s-search-bar"
            >
              <input
                className="s-search-input"
                name="search"
                onChange={[Function]}
                placeholder="Search here..."
                type="text"
                value=""
              />
              <div
                className="s-search-icon"
              />
            </div>
            <select
              className="s-location-dropdown"
              name="localeSelected"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                --Select locale--
              </option>
              <option
                value="Denver"
              >
                Denver
              </option>
              <option
                value="Remote"
              >
                Remote
              </option>
            </select>
          </div>
          <div
            className="s-filter-btn-container"
          >
            <button
              className="s-filter-btn"
              name="preferencesClicked"
              onClick={[Function]}
            >
              Filter by Preferences
            </button>
            <button
              className="s-filter-btn"
              name="allMentorsClicked"
              onClick={[Function]}
            >
              All Mentors
            </button>
            <img
              alt="favorite button"
              className="s-favorite-btn"
              name="favClicked"
              onClick={[Function]}
              src="heart.svg"
            />
          </div>
        </div>
        <Connect(Preferences)
          preferencesClicked={false}
        />
      </div>,
      "className": "s-search-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="s-search-and-buttons"
          >
            <div
              className="s-search-inputs-container"
            >
              <div
                className="s-search-bar"
              >
                <input
                  className="s-search-input"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here..."
                  type="text"
                  value=""
                />
                <div
                  className="s-search-icon"
                />
              </div>
              <select
                className="s-location-dropdown"
                name="localeSelected"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  --Select locale--
                </option>
                <option
                  value="Denver"
                >
                  Denver
                </option>
                <option
                  value="Remote"
                >
                  Remote
                </option>
              </select>
            </div>
            <div
              className="s-filter-btn-container"
            >
              <button
                className="s-filter-btn"
                name="preferencesClicked"
                onClick={[Function]}
              >
                Filter by Preferences
              </button>
              <button
                className="s-filter-btn"
                name="allMentorsClicked"
                onClick={[Function]}
              >
                All Mentors
              </button>
              <img
                alt="favorite button"
                className="s-favorite-btn"
                name="favClicked"
                onClick={[Function]}
                src="heart.svg"
              />
            </div>
          </div>,
          <Connect(Preferences)
            preferencesClicked={false}
          />,
        ],
        "className": "s-search-and-pref-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="s-search-inputs-container"
              >
                <div
                  className="s-search-bar"
                >
                  <input
                    className="s-search-input"
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here..."
                    type="text"
                    value=""
                  />
                  <div
                    className="s-search-icon"
                  />
                </div>
                <select
                  className="s-location-dropdown"
                  name="localeSelected"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    --Select locale--
                  </option>
                  <option
                    value="Denver"
                  >
                    Denver
                  </option>
                  <option
                    value="Remote"
                  >
                    Remote
                  </option>
                </select>
              </div>,
              <div
                className="s-filter-btn-container"
              >
                <button
                  className="s-filter-btn"
                  name="preferencesClicked"
                  onClick={[Function]}
                >
                  Filter by Preferences
                </button>
                <button
                  className="s-filter-btn"
                  name="allMentorsClicked"
                  onClick={[Function]}
                >
                  All Mentors
                </button>
                <img
                  alt="favorite button"
                  className="s-favorite-btn"
                  name="favClicked"
                  onClick={[Function]}
                  src="heart.svg"
                />
              </div>,
            ],
            "className": "s-search-and-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="s-search-bar"
                  >
                    <input
                      className="s-search-input"
                      name="search"
                      onChange={[Function]}
                      placeholder="Search here..."
                      type="text"
                      value=""
                    />
                    <div
                      className="s-search-icon"
                    />
                  </div>,
                  <select
                    className="s-location-dropdown"
                    name="localeSelected"
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value=""
                    >
                      --Select locale--
                    </option>
                    <option
                      value="Denver"
                    >
                      Denver
                    </option>
                    <option
                      value="Remote"
                    >
                      Remote
                    </option>
                  </select>,
                ],
                "className": "s-search-inputs-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="s-search-input"
                        name="search"
                        onChange={[Function]}
                        placeholder="Search here..."
                        type="text"
                        value=""
                      />,
                      <div
                        className="s-search-icon"
                      />,
                    ],
                    "className": "s-search-bar",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "s-search-input",
                        "name": "search",
                        "onChange": [Function],
                        "placeholder": "Search here...",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "s-search-icon",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value=""
                      >
                        --Select locale--
                      </option>,
                      <option
                        value="Denver"
                      >
                        Denver
                      </option>,
                      <option
                        value="Remote"
                      >
                        Remote
                      </option>,
                    ],
                    "className": "s-location-dropdown",
                    "name": "localeSelected",
                    "onChange": [Function],
                    "value": "",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "--Select locale--",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": "--Select locale--",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Denver",
                        "value": "Denver",
                      },
                      "ref": null,
                      "rendered": "Denver",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Remote",
                        "value": "Remote",
                      },
                      "ref": null,
                      "rendered": "Remote",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="s-filter-btn"
                    name="preferencesClicked"
                    onClick={[Function]}
                  >
                    Filter by Preferences
                  </button>,
                  <button
                    className="s-filter-btn"
                    name="allMentorsClicked"
                    onClick={[Function]}
                  >
                    All Mentors
                  </button>,
                  <img
                    alt="favorite button"
                    className="s-favorite-btn"
                    name="favClicked"
                    onClick={[Function]}
                    src="heart.svg"
                  />,
                ],
                "className": "s-filter-btn-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Filter by Preferences",
                    "className": "s-filter-btn",
                    "name": "preferencesClicked",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Filter by Preferences",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "All Mentors",
                    "className": "s-filter-btn",
                    "name": "allMentorsClicked",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "All Mentors",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "favorite button",
                    "className": "s-favorite-btn",
                    "name": "favClicked",
                    "onClick": [Function],
                    "src": "heart.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "preferencesClicked": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="s-search-and-pref-container"
        >
          <div
            className="s-search-and-buttons"
          >
            <div
              className="s-search-inputs-container"
            >
              <div
                className="s-search-bar"
              >
                <input
                  className="s-search-input"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here..."
                  type="text"
                  value=""
                />
                <div
                  className="s-search-icon"
                />
              </div>
              <select
                className="s-location-dropdown"
                name="localeSelected"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  --Select locale--
                </option>
                <option
                  value="Denver"
                >
                  Denver
                </option>
                <option
                  value="Remote"
                >
                  Remote
                </option>
              </select>
            </div>
            <div
              className="s-filter-btn-container"
            >
              <button
                className="s-filter-btn"
                name="preferencesClicked"
                onClick={[Function]}
              >
                Filter by Preferences
              </button>
              <button
                className="s-filter-btn"
                name="allMentorsClicked"
                onClick={[Function]}
              >
                All Mentors
              </button>
              <img
                alt="favorite button"
                className="s-favorite-btn"
                name="favClicked"
                onClick={[Function]}
                src="heart.svg"
              />
            </div>
          </div>
          <Connect(Preferences)
            preferencesClicked={false}
          />
        </div>,
        "className": "s-search-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="s-search-and-buttons"
            >
              <div
                className="s-search-inputs-container"
              >
                <div
                  className="s-search-bar"
                >
                  <input
                    className="s-search-input"
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here..."
                    type="text"
                    value=""
                  />
                  <div
                    className="s-search-icon"
                  />
                </div>
                <select
                  className="s-location-dropdown"
                  name="localeSelected"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    --Select locale--
                  </option>
                  <option
                    value="Denver"
                  >
                    Denver
                  </option>
                  <option
                    value="Remote"
                  >
                    Remote
                  </option>
                </select>
              </div>
              <div
                className="s-filter-btn-container"
              >
                <button
                  className="s-filter-btn"
                  name="preferencesClicked"
                  onClick={[Function]}
                >
                  Filter by Preferences
                </button>
                <button
                  className="s-filter-btn"
                  name="allMentorsClicked"
                  onClick={[Function]}
                >
                  All Mentors
                </button>
                <img
                  alt="favorite button"
                  className="s-favorite-btn"
                  name="favClicked"
                  onClick={[Function]}
                  src="heart.svg"
                />
              </div>
            </div>,
            <Connect(Preferences)
              preferencesClicked={false}
            />,
          ],
          "className": "s-search-and-pref-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="s-search-inputs-container"
                >
                  <div
                    className="s-search-bar"
                  >
                    <input
                      className="s-search-input"
                      name="search"
                      onChange={[Function]}
                      placeholder="Search here..."
                      type="text"
                      value=""
                    />
                    <div
                      className="s-search-icon"
                    />
                  </div>
                  <select
                    className="s-location-dropdown"
                    name="localeSelected"
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value=""
                    >
                      --Select locale--
                    </option>
                    <option
                      value="Denver"
                    >
                      Denver
                    </option>
                    <option
                      value="Remote"
                    >
                      Remote
                    </option>
                  </select>
                </div>,
                <div
                  className="s-filter-btn-container"
                >
                  <button
                    className="s-filter-btn"
                    name="preferencesClicked"
                    onClick={[Function]}
                  >
                    Filter by Preferences
                  </button>
                  <button
                    className="s-filter-btn"
                    name="allMentorsClicked"
                    onClick={[Function]}
                  >
                    All Mentors
                  </button>
                  <img
                    alt="favorite button"
                    className="s-favorite-btn"
                    name="favClicked"
                    onClick={[Function]}
                    src="heart.svg"
                  />
                </div>,
              ],
              "className": "s-search-and-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="s-search-bar"
                    >
                      <input
                        className="s-search-input"
                        name="search"
                        onChange={[Function]}
                        placeholder="Search here..."
                        type="text"
                        value=""
                      />
                      <div
                        className="s-search-icon"
                      />
                    </div>,
                    <select
                      className="s-location-dropdown"
                      name="localeSelected"
                      onChange={[Function]}
                      value=""
                    >
                      <option
                        value=""
                      >
                        --Select locale--
                      </option>
                      <option
                        value="Denver"
                      >
                        Denver
                      </option>
                      <option
                        value="Remote"
                      >
                        Remote
                      </option>
                    </select>,
                  ],
                  "className": "s-search-inputs-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="s-search-input"
                          name="search"
                          onChange={[Function]}
                          placeholder="Search here..."
                          type="text"
                          value=""
                        />,
                        <div
                          className="s-search-icon"
                        />,
                      ],
                      "className": "s-search-bar",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "s-search-input",
                          "name": "search",
                          "onChange": [Function],
                          "placeholder": "Search here...",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "s-search-icon",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "div",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <option
                          value=""
                        >
                          --Select locale--
                        </option>,
                        <option
                          value="Denver"
                        >
                          Denver
                        </option>,
                        <option
                          value="Remote"
                        >
                          Remote
                        </option>,
                      ],
                      "className": "s-location-dropdown",
                      "name": "localeSelected",
                      "onChange": [Function],
                      "value": "",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "--Select locale--",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": "--Select locale--",
                        "type": "option",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Denver",
                          "value": "Denver",
                        },
                        "ref": null,
                        "rendered": "Denver",
                        "type": "option",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Remote",
                          "value": "Remote",
                        },
                        "ref": null,
                        "rendered": "Remote",
                        "type": "option",
                      },
                    ],
                    "type": "select",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="s-filter-btn"
                      name="preferencesClicked"
                      onClick={[Function]}
                    >
                      Filter by Preferences
                    </button>,
                    <button
                      className="s-filter-btn"
                      name="allMentorsClicked"
                      onClick={[Function]}
                    >
                      All Mentors
                    </button>,
                    <img
                      alt="favorite button"
                      className="s-favorite-btn"
                      name="favClicked"
                      onClick={[Function]}
                      src="heart.svg"
                    />,
                  ],
                  "className": "s-filter-btn-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Filter by Preferences",
                      "className": "s-filter-btn",
                      "name": "preferencesClicked",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Filter by Preferences",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "All Mentors",
                      "className": "s-filter-btn",
                      "name": "allMentorsClicked",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "All Mentors",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "favorite button",
                      "className": "s-favorite-btn",
                      "name": "favClicked",
                      "onClick": [Function],
                      "src": "heart.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "preferencesClicked": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
